### PR TITLE
Enable labeling datetime plots

### PR DESCRIPTION
**Pull request by nateGeorge**
It turns out all that was needed to label datetime plots was to properly check if numpy arrays are datetimes, then convert them to floats with date2num. I added some extra boolean tests and changed the name of the function in `labelLine` from `x_to_float` to `ensure_float`, and applied it to a few extra places where needed.

I tried running the tests, but some of them failed for some reason. However, when I ran the code in the tests by themselves, they worked, so I'm not sure why they failed. It was saying things like "overflow error".

This addresses issue #15.

### CODE REVIEW

File: labellines/core.py [hunk 1]
```diff
@@ -26,8 +26,28 @@ def labelLine(line, x, label=None, align=True, drop_label=False, **kwargs):
...
+            if isinstance(value, datetime) \
+            or isinstance(value, np.datetime64) \
+            or np.issubdtype(value.dtype, np.datetime64) \
+            or str(value.dtype).startswith("datetime64") \
+            or value.dtype == 'O':
+                return date2num(value)
+            else:  # another numpy dtype like float64
+                return value
+        except AttributeError:  # possibly int or other float/int dtype
+            return value
```
Comment (cphyc) [inline labellines/core.py:46]: Could this be simplified as follows?
```suggestion
        try:
            return date2num(value)
        except AttributeError:  # possibly int or other float/int dtype
            return value
```

**Review by cphyc — APPROVED** [review]
This looks great, thanks for the suggestion!

File: labellines/core.py (hunk 1 above)
Comment (nateGeorge) [reply labellines/core.py:46]: Possibly, we would need to test it. I think the various dtypes/classes we're catching there are datetime.datetime, and np timestamps with/without timezones.

Comment (cphyc) [reply labellines/core.py:46]: Would you mind trying and see whether the tests pass?

Comment (nateGeorge) [reply labellines/core.py:46]: I put it on my todo list. Hopefully this weekend.

**Pull request merged**